cmd/subnetcmd/join.go

```diff
@@ -67,18 +78,101 @@ func joinCmd(cmd *cobra.Command, args []string) error {
 	network = models.NetworkFromString(networkStr)
 	networkLower := strings.ToLower(network.String())
 
-	if avagoConfigPath == "" {
+	ask := "Would you like to check if your node is already whitelisted to join this subnet?"
+	yes, err := prompts.CaptureYesNo(ask)
+	if err != nil {
+		return err
+	}
+	if yes {
+		isWhitelisted, err := checkIsWhitelisted(ids.GenerateTestID(), network)
+		if err != nil {
+			return err
+		}
+		if !isWhitelisted {
+			ux.Logger.PrintToUser(`The node seems to not have been whitelisted. 
+You can continue with this command, generating a config file or printing the whitelisting configuration,
+but until the node is not whitelisted, it will not be able to validate this subnet.`)
+			y, err := prompts.CaptureYesNo("Do you wish to continue")
+			if err != nil {
+				return err
+			}
+			if !y {
+				return nil
+			}
+		}
+	}
+
+	if print {
 		printJoinCmd(subnetIDstr, networkLower)
-	} else {
-		if err := editConfigFile(subnetIDstr, networkLower); err != nil {
+		return nil
+	}
+
+	if avagoConfigPath == "" {
+		const (
+			choiceManual    = "Manual"
+			choiceAutomatic = "Automatic"
+		)
+		choice, err := prompts.CaptureList(
+			"How would you like to update the avalanchego config?",
+			[]string{choiceManual, choiceAutomatic},
+		)
+		if err != nil {
 			return err
 		}
+		switch choice {
+		case choiceManual:
+			printJoinCmd(subnetIDstr, networkLower)
+			return nil
+		case choiceAutomatic:
+			avagoConfigPath, err = prompts.CaptureString("Path to your existing config file (or where it will be generated)")
+			if err != nil {
+				return err
+			}
+		}
+		// if choice is automatic, we just pass through this block,
+		// so we don't need another else if the the config path is not set
+	}
+	if err := editConfigFile(subnetIDstr, networkLower); err != nil {
+		return err
 	}
 	return nil
 }
 
+func checkIsWhitelisted(subnetID ids.ID, network models.Network) (bool, error) {
+	/*
+		promptStr := "Please enter your node's ID (NodeID-...)"
+		nodeID, err := prompts.CaptureNodeID(promptStr)
+		if err != nil {
+			return false, err		}
+	*/
+	nodeID := ids.GenerateTestNodeID()
+	var api string
+	switch network {
```

Comment:
do we want to move this to its own function? I imagine we will need to use this logic a bunch

Reply:
That is a good idea but I suggest to move this down the line a bit: Many functions are currently very network-dependent, meaning some are fuji only, some local only, and thus what the `switch` is allowed to contain is not (yet?) consistent.